src/amo/components/HeroRecommendation/index.js

```diff
@@ -0,0 +1,122 @@
+/* @flow */
+import invariant from 'invariant';
+import * as React from 'react';
+import { compose } from 'redux';
+
+import translate from 'core/i18n/translate';
+import type { I18nType } from 'core/types/i18n';
+
+import promoImage from './img/christin-hume-mfB1B1s4sMc-unsplash.png';
+import './styles.scss';
+
+type Props = {|
+  body: string,
+  heading: string,
+  linkText: string,
+  linkHref: string,
+|};
+
+type InternalProps = {|
+  ...Props,
+  i18n: I18nType,
+|};
+
+export class HeroRecommendationBase extends React.Component<InternalProps> {
+  renderOverlayShape() {
+    const gradientA = 'HeroRecommendation-gradient-a';
+    const gradientB = 'HeroRecommendation-gradient-b';
+
+    return (
+      <svg
+        className="HeroRecommendation-overlayShape"
+        xmlns="http://www.w3.org/2000/svg"
+        viewBox="0 0 334 307"
+      >
+        <defs>
+          <linearGradient
+            id={gradientA}
+            x1="37.554%"
+            x2="20.41%"
+            y1="51.425%"
+            y2="49.159%"
+          >
+            <stop
+              className="HeroRecommendation-gradientA-startColor"
+              offset="0%"
+            />
+            <stop
+              className="HeroRecommendation-gradientA-endColor"
+              offset="100%"
+            />
+          </linearGradient>
+          <linearGradient
+            id={gradientB}
+            x1="0%"
+            x2="100%"
+            y1="36.092%"
+            y2="36.092%"
+          >
+            <stop
+              className="HeroRecommendation-gradientB-startColor"
+              offset="0%"
+            />
+            <stop
+              className="HeroRecommendation-gradientB-endColor"
+              offset="100%"
+            />
+          </linearGradient>
+        </defs>
+        <g fill="none">
+          <path
+            className="HeroRecommendation-solidSwoosh"
+            d="M0 307h267c-52.113-48.37-94.22-112.103-120.226-188.126C122.438 47.736 63.457 3.044 0 0v307z"
+          />
+          <path
+            fill={`url(#${gradientA})`}
+            d="M0 307c17.502-9.934 34.574-21.458 51.072-34.602C121.246 216.49 216.932 232.615 271 307H0z"
+            transform="matrix(-1 0 0 1 334 0)"
+          />
+          <path
+            fill={`url(#${gradientB})`}
+            d="M334 307H114c31.79-24.866 61.545-54.258 88.533-88.135C237.964 174.386 285.12 148.906 334 143v164z"
+            transform="matrix(-1 0 0 1 334 0)"
+          />
+        </g>
+      </svg>
+    );
+  }
+
+  render() {
+    const { body, heading, i18n, linkText, linkHref } = this.props;
+
+    invariant(body, 'The body property is required');
+    invariant(heading, 'The heading property is required');
+    invariant(linkText, 'The linkText property is required');
+    invariant(linkHref, 'The linkHref property is required');
```

Comment:
I thought we were not doing this because we had React proptypes validation already (_via_ flow). Is that a new thing?

Reply:
[pages/Home](https://github.com/mozilla/addons-frontend/blob/master/src/amo/pages/Home/index.js), which instantiates this component, does not use Flow 😞

Reply:
ah, but react built-in prop-types validation should still work, no? It's probably fine to leave those `invariant()` calls here but please add a comment so that we remember that it's because `Home` does not use Flow and that we should remove these calls once we convert `Home` to Flow

Reply:
No, prop-type linting does not understand Flow. If I remove a required prop from `Home`'s call to this component, there are no lint errors.

> add a comment so that we remember that it's because `Home` does not use Flow

It's not just `Home`, it could be any non-Flow component now or in the future. I don't think it's safe to build any component at all without `invariant()` calls until `addons-frontend` is 100% Flow.

Reply:
I see. I thought we had something to use Flow annotations with React prop-types actually.